drivers/serial/uart_nxp_s32_linflexd.c

```diff
@@ -362,6 +401,17 @@ static DEVICE_API(uart, uart_nxp_s32_driver_api) = {
 	}
 
 #define UART_NXP_S32_INIT_DEVICE(n)						\
+	BUILD_ASSERT(DT_INST_ENUM_IDX(n, stop_bits) == UART_CFG_STOP_BITS_1 ||	\
```

Comment:
why not overriding the binding defaults in `nxp,s32-linflexd.yaml`, e.g. allowing `stop-bits` to be `enum: ["1", "2"]`?
Unless there's a dts guideline that says this is not recommended.. but I don't see any in https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html

Reply:
The binding does not allow overwriting the default enum values of a property defined in the included binding `uart-controller.yaml`. If try to overwrite , an error will occur:
![image](https://github.com/user-attachments/assets/2d65fa31-e94b-43d9-a791-bddb58c8640c)

Or you are suggesting duplicating the properties already defined in `uart-controller.yaml` and remove include `uart-controller.yaml`?

Other point, I want the value enum binding index must to be define to correspond with the value enum configuration in [uart.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/uart.h#L87). Due to it's used to convert to the value hw config.